src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor_code.go

```diff
@@ -180,7 +180,7 @@ func (a *StoragePowerActorCode_I) OnMinerSurprisePoStFailure(rt Runtime, numCons
 	}
 }
 
-func (a *StoragePowerActorCode_I) ReportConsensusFault(rt Runtime, slasherAddr addr.Address, faultType ConsensusFaultType, proof []block.Block) {
+func (a *StoragePowerActorCode_I) ReportConsensusFault(rt Runtime, faultType ConsensusFaultType, proof []block.BlockHeader) {
```

Comment:
I hope that we can remove visibility of the BlockHeader type here and instead delegate the fault checking to the runtime. I'll file an issue for it.

Reply:
the check here was supposed to be done in ExpectedConsensus. Does this mean that that logic will now also live in the runtime? I am inclined to leave this as a follow up, there might be other places in actor that require similar validation cc @sternhenri